core/src/avm2/globals/flash/display/shader_job.rs

```diff
@@ -0,0 +1,165 @@
+use ruffle_render::{
+    bitmap::PixelRegion,
+    pixel_bender::{
+        PixelBenderParam, PixelBenderParamQualifier, PixelBenderShaderArgument, PixelBenderType,
+        OUT_COORD_NAME,
+    },
+};
+
+use crate::{
+    avm2::{string::AvmString, Activation, Error, Object, TObject, Value},
+    avm2_stub_method,
+    pixel_bender::PixelBenderTypeExt,
+};
+
+/// Implements `ShaderJob.start`.
+pub fn start<'gc>(
+    activation: &mut Activation<'_, 'gc>,
+    this: Option<Object<'gc>>,
+    _args: &[Value<'gc>],
+) -> Result<Value<'gc>, Error<'gc>> {
+    let this = this.unwrap();
+
+    avm2_stub_method!(
+        activation,
+        "flash.display.ShaderJob",
+        "start",
+        "async execution and non-BitmapData inputs"
+    );
+
+    let shader = this
+        .get_public_property("shader", activation)?
+        .as_object()
+        .expect("Missing Shader object");
+    let shader_data = shader
+        .get_public_property("data", activation)?
+        .as_object()
+        .expect("Missing ShaderData object")
+        .as_shader_data()
+        .expect("ShaderData object is not a ShaderData instance");
```

Comment:
Since these could be `null`, would it be better to do null handling rather than panic?

Reply:
This PR is already very large, so I'm postponing nicer error handling for a follow-up PR.

Reply:
Could you add some TODO notices in that case?